app/src/main/java/io/appium/uiautomator2/core/InteractionController.java

```diff
@@ -62,48 +67,92 @@ public boolean injectEventSync(final InputEvent event) throws UiAutomator2Except
         return injectEventSync(event, true);
     }
 
+    public boolean shouldTrackScrollEvents() {
+        Session session = AppiumUIA2Driver.getInstance().getSessionOrThrow();
+        Boolean trackScrollEvents = true;
+        if (session.hasCapability(TRACK_SCROLL_EVENT_CAP)) {
+          try {
+              trackScrollEvents = (Boolean) session.getCapability(TRACK_SCROLL_EVENT_CAP);
```

Comment:
how about using String.format?

Reply:
same below

Reply:
I personally think the `else` ought to be there when it is, logically, an else, even if the `return` makes it technically "redundant".